src/Series.cpp

```diff
@@ -49,6 +50,20 @@ namespace
      */
     std::string cleanFilename(std::string const &filename, Format f);
 
+    struct Match
+    {
+        bool isContained{};
+        int padding{};
+        uint64_t iteration{};
```

Comment:
```suggestion
        bool isContained{}; //! pattern match successful
        int padding{}; //! number of zeros used for padding of iteration
        uint64_t iteration{}; //! iteration found in regex pattern (default: 0)
```
To avoid confusion, one could make iteration an `int64_t` and return `-1` by default (or so). But we already use `uint64_t` everywhere, so that's fine, too.

Reply:
If we want to do this, I think the cleanest way would be by combining `isContained` and `iteration` into an `auxiliary::Option<uint64_t>`. Should we use something like this right away?